### PR TITLE
Wip/11053 direct read write on device

### PR DESCRIPTION
**Pull request by vaeng**
This PR also includes the code additions to support the device catalogue from task #7673.

### CODE REVIEW

File: PythonModuleMethods.h [hunk 1]
```diff
@@ -8,12 +8,18 @@
...
+#include <ChimeraTK/RegisterInfo.h>
+#include <ChimeraTK/RegisterCatalogue.h>
 
 namespace p = boost::python;
 namespace np = boost::python::numpy;
 
 namespace mtca4upy {
 
+
+  //ChimeraTK::RegisterInfo getRegister(const std::string& registerPathName);
```
Comment (mhier) [inline PythonModuleMethods.h:20]: No commented code please...

**Review by mhier — CHANGES_REQUESTED** [review]
The changes itself look good. I have only a formal comment: The C++ code does not follow our formatting rules. Please run clang-format to fix that.

**Comment by vaeng** [comment]
Good catch @mhier ! I had to re-setup clang-format. Does that look correct now?

**Review by mhier — APPROVED** [review]
Thanks for the reformatting. There is one comment, but I already approve to speed things up. If this TODO si for the future, please create a ticket and link to it in the source code comment.

File: deviceaccessPython.cc [hunk 2]
```diff
@@ -16,7 +19,7 @@
   FUNCTION_TEMPLATE(uint64_t, func_name, _uint64)                                                                      \
   FUNCTION_TEMPLATE(float, func_name, _float)                                                                          \
   FUNCTION_TEMPLATE(double, func_name, _double)
-/* TODO: Implement String and Bool Type 
+/* TODO: Implement String and Bool Type
```
[the remark this answers is not in the document]
Comment (vaeng) [reply deviceaccessPython.cc:22]: That is for the future and not in the scope of this feature. I have to take the time and write a conversation for the usertypes of Void, Boolean, and string. So far it does not seem like anyone is using these.

**Pull request merged**